### PR TITLE
feat: add workflow to format Lua files in pull requests

### PR DESCRIPTION
**Pull request by omarcopires**
**Description**
This pull request adds a workflow to the project that automatically formats Lua files in pull requests. The goal is to ensure that all Lua files are formatted according to established guidelines, promoting consistency and improving code readability.

**What was done**
- Added a formatting workflow for Lua files.
- Configured to run automatically on pull requests.
- The workflow uses [formatting tool] to apply the standard formatting.

**How to test**
- Create a pull request with Lua files that are not correctly formatted.
- Verify that the formatting workflow runs automatically.
- Ensure that Lua files are formatted according to the established guidelines.

**Notes**
- Ensure that the formatting tool is correctly configured in your local environment to avoid discrepancies.
- Formatting will be applied automatically to code before merging into the main branch.

### CODE REVIEW

**Comment by omarcopires** [comment]
just to maintain a standard in the lua code as well, considering that the source already has a defined standard.
I don't have the necessary knowledge to change the format or any other modification to the workflow, so I left the pull request free for administrators to make changes if necessary!

File: data/actions/lib/actions.lua [hunk 1]
```diff
@@ -1,20 +1,71 @@
...
-local sandIds = {231, 9059} -- desert sand
+local groundIds = { 354, 355 } -- pick usable ground
+local sandIds = { 231, 9059 } -- desert sand
 local holeId = { -- usable rope holes, for rope spots see global.lua
-	294, 369, 370, 383, 392, 408, 409, 410, 427, 428, 429, 430, 462, 469, 470, 482,
-	484, 485, 489, 924, 1369, 3135, 3136, 4835, 4837, 7933, 7938, 8170, 8249, 8250,
-	8251, 8252, 8254, 8255, 8256, 8276, 8277, 8279, 8281, 8284, 8285, 8286, 8323,
-	8567, 8585, 8595, 8596, 8972, 9606, 9625, 13190, 14461, 19519, 21536, 23713,
-	26020
+	294,
```
Comment (EPuncker) [inline data/actions/lib/actions.lua:10]: noooooooooo, one id per line is ugly

Comment (omarcopires) [reply data/actions/lib/actions.lua:10]: There must be other formatting styles that can be applied, I have no doubt! (I agree that this is very ugly).

**Comment by ranisalt** [comment]
I don't like the idea of manipulating others' pull requests because it causes surprise that your PR was modified by a third party. Also, people are generally awful at using Github :sweat_smile: 

I'd rather add a pre-commit hook that formats changed Lua files so they are pushed as formatted, and then we just check formatting in a workflow

**Pull request closed without merging**